actors/miner/tests/withdraw_balance.rs

```diff
@@ -66,6 +70,141 @@ fn withdraw_only_what_we_can_after_fee_debt() {
 
     let requested = rt.balance.borrow().to_owned();
     let expected_withdraw = &requested - &fee_debt;
-    h.withdraw_funds(&mut rt, &requested, &expected_withdraw, &fee_debt).unwrap();
+    h.withdraw_funds(&mut rt, h.owner, &requested, &expected_withdraw, &fee_debt).unwrap();
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_withdraw() {
+    let mut h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.set_balance(TokenAmount::from(BIG_BALANCE));
+    h.construct_and_verify(&mut rt);
+
+    let one = TokenAmount::from(1);
+    h.withdraw_funds(&mut rt, h.owner, &one, &one, &TokenAmount::zero()).unwrap();
+
+    let first_beneficiary_id = Address::new_id(999);
+    let quota = TokenAmount::from(ONE_PERCENT_BALANCE);
+    h.propose_approve_initial_beneficiary(
+        &mut rt,
+        first_beneficiary_id,
+        BeneficiaryTerm::new(quota, TokenAmount::zero(), PERIOD_OFFSET + 100),
+    )
+    .unwrap();
+    h.withdraw_funds(&mut rt, h.owner, &one, &one, &TokenAmount::zero()).unwrap();
+    h.withdraw_funds(&mut rt, h.beneficiary, &one, &one, &TokenAmount::zero()).unwrap();
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_withdraw_allow_zero() {
+    let mut h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.set_balance(TokenAmount::from(BIG_BALANCE));
+    h.construct_and_verify(&mut rt);
+
+    let first_beneficiary_id = Address::new_id(999);
+    h.propose_approve_initial_beneficiary(
+        &mut rt,
+        first_beneficiary_id,
+        BeneficiaryTerm::new(TokenAmount::from(1), TokenAmount::zero(), PERIOD_OFFSET + 100),
+    )
+    .unwrap();
+    h.withdraw_funds(
+        &mut rt,
+        first_beneficiary_id,
+        &TokenAmount::zero(),
+        &TokenAmount::zero(),
+        &TokenAmount::zero(),
+    )
+    .unwrap();
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_withdraw_more_than_quota() {
```

Comment:
Maybe `successfully_withdraw_limited_to_quota`? It requests more than quota, but doesn't get it.